File: SampleSolution.Core/Services/Base/CrudService.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;
using SampleSolution.Core.Models.Entities.Base;
using SampleSolution.Core.Models.ViewModels.Pagination;
using SampleSolution.Core.Repositories.Base;

namespace SampleSolution.Core.Services.Base
{
    public class CrudService<TEntity, TId> : ICrudService<TEntity, TId>
        where TEntity : class, IEntity<TId>, new()
    {
        protected ICrudRepository<TEntity, TId> Repository { get; }

        public CrudService(ICrudRepository<TEntity, TId> repository)
        {
            this.Repository = repository;
        }

        public async Task<TEntity> Create(TEntity entity)
        {
            return await this.Repository.Create(entity);
        }

        public async Task Delete(TId id)
        {
            await this.Repository.Delete(id);
        }

        public async Task<TEntity> GetById(TId id)
        {
            return await this.Repository.GetById(id);
        }

        public async Task<IList<TEntity>> GetAll()
        {
            return await this.Repository.GetAll();
        }

        public async Task Update(TId id, TEntity entity)
        {
            await this.Repository.Update(id, entity);
        }

        public async Task<PagedList<TEntity>> GetPaged(PagedListQuery query)
        {
            return await this.Repository.GetPaged(query);
        }

        public async Task<TEntity> GetGraph(TId id, JObject graph)
        {
            return await this.Repository.GetGraph(id, graph);
        }
    }
}

File: Clients/SampleSolution.Web/wwwrootsrc/core/scripts/app/services/_base/CRUDService.ts
import {BaseService} from './BaseService';
import {CRUDRepository} from '../../repositories/_base/CRUDRepository';
import {Serializable} from 'ts-serializable';
import {PageListQuery} from '../../models/ViewModels/PageListQuery';
import {PagedList} from '../../models/ViewModels/PagedList';
import {IGraphQuery} from '../../models/ViewModels/GraphQuery';

export class CRUDService<T1 extends Serializable, T2 extends CRUDRepository<T1>> extends BaseService {

    public repository: T2;

    constructor(repository: new () => T2) {
        super();
        this.repository = new repository();
    }

    public async getAll(...keys: (string | number)[]): Promise<T1[]> {
        return await this.repository.getAll(...keys);
    }

    public async create(value: T1, ...key: (string | number)[]): Promise<T1> {
        return await this.repository.create(value, ...key);
    }

    public async getById(id: string | number, ...key: (string | number)[]): Promise<T1> {
        return await this.repository.getById(id, ...key);
    }

    public async getGraphById(id: string | number, ...key: (string | number | IGraphQuery)[]): Promise<T1> {
        return await this.repository.getGraphById(id, ...key);
    }

    public async update(value: T1, ...key: (string | number)[]): Promise<void> {
        return await this.repository.update(value, ...key);
    }

    public async delete(...key: (string | number)[]): Promise<void> {
        return await this.repository.delete(...key);
    }

    public async getPaged(...keys: (string | number | PageListQuery)[]): Promise<PagedList<T1>> {
        return await this.repository.getPaged(...keys);
    }

}
